design/sketch.webidl

```diff
@@ -465,35 +495,12 @@ interface WebGPUCommandEncoder {
     void copyTextureToBuffer();
     void copyTextureToTexture();
     void blit();
+};
 
-    void transitionBuffer(WebGPUBuffer b, WebGPUBufferUsageFlags f);
-
-    // Allowed in both compute and render passes
-    void setPushConstants(WebGPUShaderStageFlags stage,
-                          u32 offset,
-                          u32 count,
-                          ArrayBuffer data);
-    void setBindGroup(u32 index, WebGPUBindGroup bindGroup);
-    void setPipeline((WebGPUComputePipeline or WebGPURenderPipeline) pipeline);
-
-    // Compute pass commands
-    void beginComputePass();
-    void endComputePass();
-
-    void dispatch(u32 x, u32 y, u32 z);
-
-    // Render pass commands
-    void beginRenderPass(WebGPURenderPassDescriptor descriptor);
-    void endRenderPass();
-
-    void setBlendColor(float r, float g, float b, float a);
-    void setIndexBuffer(WebGPUBuffer buffer, u32 offset);
-    void setVertexBuffers(u32 startSlot, sequence<WebGPUBuffer> buffers, sequence<u32> offsets);
-
-    void draw(u32 vertexCount, u32 instanceCount, u32 firstVertex, u32 firstInstance);
-    void drawIndexed(u32 indexCount, u32 instanceCount, u32 firstIndex, u32 firstInstance, u32 firstVertex);
-
-    // TODO add missing commands
+dictionary WebGPUCommandBufferDescriptor {
+    /// If true, the command buffer can be submitted for execution any number
+    /// of times. Otherwise, it's invalidated after the first submit.
+    bool reusable;
```

Comment:
I don’t think this is implementable on Metal.

Reply:
Correct. It's not directly implementable on Metal. And we should totally discuss this specifically on the call.
Meanwhile, I can think of the following points towards supporting the feature:
  1. having this flag was part of the reason why `WebGPUCommandBuffer` as a concept makes sense (as opposed to just a list of passes)
  2. Google's "Dawn" would have no trouble implementing this on Metal, since they record command buffers at submission time. Mozilla's implementation is already targeting Vulkan, so re-using command buffers has been available for a long time. I imagine Apple's implementation would have a problem with this feature. Is this the case?